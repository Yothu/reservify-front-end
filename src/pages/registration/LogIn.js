import { toast } from 'react-toastify';
import React, { useEffect } from 'react';
import { useForm } from 'react-hook-form';
import { Oval } from 'react-loader-spinner';
import { useNavigate } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { userLogin, reset } from '../../redux/auth/auth-slice';
import loginImg from '../../assets/images/undraw-svg/green-welcome.svg';
import ErrorMsg from '../../components/Ui/ErrorMsg';
import Button from '../../components/Ui/Button';

function LogIn() {
  const {
    register,
    handleSubmit,
    formState: { errors },
    // watch,
  } = useForm({
    defaultValues: {
      email: '',
      password: '',
    },
  });

  // prettier-ignore
  const {
    user, isLoading, isSuccess, isError, message,
  } = useSelector(
    (state) => state.auth,
  );

  const dispatch = useDispatch();
  const navigate = useNavigate();

  useEffect(() => {
    if (isError) {
      toast.error(message);
    }
    if (isSuccess || user) {
      toast.success('Successfully logged in');
      navigate('/main');
    }
    dispatch(reset());
  }, [user, isError, isSuccess, message, navigate, dispatch]);

  const onSubmit = (data) => {
    dispatch(userLogin(data));
  };

  if (isLoading) {
    return (
      <div className="d-flex justify-content-center">
        <Oval color="#FBBC05" height={250} width={250} />
      </div>
    );
  }

  return (
    <div className="d-flex flex-column align-items-center pt-4 container">
      <div className="text-center mb-3">
        <img src={loginImg} alt="register" className="mw-100" width={400} />
      </div>
      <form onSubmit={handleSubmit(onSubmit)} className="w-100">
        <input
          type="email"
          className="form-control mb-4"
          placeholder="Enter email"
          id="email"
          {...register('email', {
            required: 'Email is required',
            pattern: {
              value: /\S+@\S+\.\S+/,
              message: 'Invalid email address',
            },
          })}
        />
        {errors.email && <ErrorMsg message={errors.email?.message} />}

        <input
          type="password"
          className="form-control my-3"
          placeholder="Enter password"
          id="password"
          {...register('password', {
            required: 'Password is required',
            minLength: {
              value: 6,
              message: 'Min length is 6',
            },
          })}
        />

        {errors.password && <ErrorMsg message={errors.password?.message} />}

        <Button type="submit" text="Login" cName="mt-3 w-100" />
      </form>
    </div>
  );
}

export default LogIn;
