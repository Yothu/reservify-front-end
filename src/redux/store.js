import { configureStore } from '@reduxjs/toolkit';
import hotelReducer from './hotels/hotel-slice';
import authReducer from './auth/auth-slice';
import reservationReducer from './reservations/reservation-slice';
import locationReducer from './locations/location-slice';

const store = configureStore({
  reducer: {
    auth: authReducer,
    hotel: hotelReducer,
    reservation: reservationReducer,
    location: locationReducer,
  },
});

export default store;
