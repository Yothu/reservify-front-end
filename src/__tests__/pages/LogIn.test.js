import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../redux/store';
import LogIn from '../../pages/registration/LogIn';

describe('Login Page', () => {
  it('renders correctly', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <LogIn />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
