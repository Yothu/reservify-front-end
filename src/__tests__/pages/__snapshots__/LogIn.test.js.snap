// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="d-flex flex-column align-items-center pt-4 container"
      >
        <div
          class="text-center mb-3"
        >
          <img
            alt="register"
            class="mw-100"
            src="green-welcome.svg"
            width="400"
          />
        </div>
        <form
          class="w-100"
        >
          <input
            class="form-control mb-4"
            id="email"
            name="email"
            placeholder="Enter email"
            type="email"
          />
          <input
            class="form-control my-3"
            id="password"
            name="password"
            placeholder="Enter password"
            type="password"
          />
          <button
            class="button mt-3 w-100"
            type="submit"
          >
            Login
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="d-flex flex-column align-items-center pt-4 container"
    >
      <div
        class="text-center mb-3"
      >
        <img
          alt="register"
          class="mw-100"
          src="green-welcome.svg"
          width="400"
        />
      </div>
      <form
        class="w-100"
      >
        <input
          class="form-control mb-4"
          id="email"
          name="email"
          placeholder="Enter email"
          type="email"
        />
        <input
          class="form-control my-3"
          id="password"
          name="password"
          placeholder="Enter password"
          type="password"
        />
        <button
          class="button mt-3 w-100"
          type="submit"
        >
          Login
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
